src/NLog/Logger.cs

```diff
@@ -95,14 +96,35 @@ public bool IsEnabled(LogLevel level)
             return this.GetTargetsForLevel(level) != null;
         }
 
+#if NET4_5 
         /// <summary>
         /// Writes the specified diagnostic message.
         /// </summary>
         /// <param name="logEvent">Log event.</param>
-        public void Log(LogEventInfo logEvent)
+        /// <param name="memberName">Method or property name of the caller.</param>
+        /// <param name="sourceFilePath">Full path of the source file that contains the caller.</param>
+        /// <param name="sourceLineNumber">Line number in the source file at which the method is called.</param>
+#else   
+        /// <summary>
+        /// Writes the specified diagnostic message.
+        /// </summary>
+        /// <param name="logEvent">Log event.</param>
+#endif
+        public void Log(LogEventInfo logEvent
+#if NET4_5
+           , [CallerMemberName] string memberName = ""
+           , [CallerFilePath] string sourceFilePath = ""
+           , [CallerLineNumber] int sourceLineNumber = 0
+#endif
```

Comment:
If you remove `Log(LogEventInfo)` and replace it with `Log(LogEventInfo, string, string, int)` method, it will cause binary incompatibility.
Consider, for example, a library compiled against NLog for .NET 4.0 that is being redirected to NLog for .NET 4.5 upon load — it will fail to find the method `Log(LogEventInfo)`